frontend/docsettings.lua

```diff
@@ -75,6 +76,16 @@ function DocSettings:getSidecarDir(doc_path, force_location)
     local location = force_location or G_reader_settings:readSetting("document_metadata_folder", "doc")
     if location == "dir" then
         path = DOCSETTINGS_DIR..path
+    elseif location == "hash" then
+        -- local doc = document:extend()
+        local file = io.open(doc_path, 'rb')
+        if not file then return '' end
```

Comment:
We prefer strings (even empty ones) with double quotes `""` , rather than your `''` here and below.
Also, undeeded/obsolete comments here and below.